src/components/user-list-item.component.js

```diff
@@ -72,81 +61,108 @@ const styles = StyleSheet.create({
   },
 });
 
-export const UserListItem = ({
-  user,
-  title,
-  subtitle,
-  titleStyle,
-  onlyImageNavigate,
-  navigation,
-  icon,
-  noBorderBottom,
-  iconAction,
-}: Props) => {
-  const ContainerComponent =
-    iconAction || onlyImageNavigate ? View : TouchableHighlight;
-  const UserComponent =
-    iconAction && !onlyImageNavigate ? TouchableOpacity : View;
-  const ImageContainerComponent = onlyImageNavigate ? TouchableOpacity : View;
-  const IconComponent = iconAction ? TouchableOpacity : View;
+const mapStateToProps = state => ({
+  authUser: state.auth.user,
+});
+
+class UserListItemComponent extends Component {
+  props: {
+    user: Object,
+    title: any,
+    subtitle: string,
+    onlyImageNavigate: boolean,
+    titleStyle: Object,
+    navigation: Object,
+    icon: string,
+    iconAction: Function,
+    noBorderBottom: boolean,
+    authUser: Object,
+  };
+
+  render() {
+    const {
+      user,
+      title,
+      subtitle,
+      titleStyle,
+      onlyImageNavigate,
+      navigation,
+      icon,
+      noBorderBottom,
+      iconAction,
+      authUser,
+    } = this.props;
+
+    const ContainerComponent =
+      iconAction || onlyImageNavigate ? View : TouchableHighlight;
+    const UserComponent =
+      iconAction && !onlyImageNavigate ? TouchableOpacity : View;
+    const ImageContainerComponent = onlyImageNavigate ? TouchableOpacity : View;
+    const IconComponent = iconAction ? TouchableOpacity : View;
 
-  return (
-    <ContainerComponent
-      onPress={() =>
-        navigation.navigate(
-          user.type === 'User' ? 'Profile' : 'Organization',
-          user.type === 'User' ? { user } : { organization: user }
-        )}
-      underlayColor={colors.greyLight}
-      style={!noBorderBottom && styles.borderContainer}
-    >
-      <View style={styles.wrapper}>
-        <UserComponent
-          style={styles.userInfo}
-          onPress={() =>
-            navigation.navigate('Profile', {
-              user,
-            })}
-        >
-          <ImageContainerComponent
-            style={styles.avatarContainer}
+    const CorrectScreen =
+      authUser.login === user.login ? 'AuthProfile' : 'Profile';
+
+    return (
+      <ContainerComponent
+        onPress={() =>
+          navigation.navigate(
+            user.type === 'User' ? CorrectScreen : 'Organization',
+            user.type === 'User' ? { user } : { organization: user }
+          )}
+        underlayColor={colors.greyLight}
+        style={!noBorderBottom && styles.borderContainer}
+      >
+        <View style={styles.wrapper}>
+          <UserComponent
+            style={styles.userInfo}
             onPress={() =>
-              navigation.navigate('Profile', {
+              navigation.navigate(CorrectScreen ? 'AuthProfile' : 'Profile', {
```

Comment:
You can just left `CorrectScreen`, because we store the name of the screen, then you do not need a ternary operator.